### PR TITLE
Add ed25519 public key bytes wrapper

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR adds a public key bytes wrapper that does not require validation to exist.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Modified a few existing tests to use the new type.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by semenov-vladyslav** [comment]
What is the purpose of such wrapper? In which way the wrapper "does not require validation"?

In my opinion, it adds extra code duplication without clear benefits. Could you please provide a non-trivial use-case example demonstrating benefits of the wrapper?

**Comment by DaughterOfMars** [comment]
It does not require validation of bytes for the type to be constructed, unlike `PublicKey`. This allows us to unpack a public key without guaranteeing anything other than the length of the bytes. We need this in shimmer because currently invalid bytes are allowed syntactically (something we hope to change for 2.0).

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "iota-crypto"
-version = "0.23.0"
+version = "0.23.1"
```
Comment (semenov-vladyslav) [inline Cargo.toml]: We do not update version manually. See [.changes](https://github.com/iotaledger/crypto.rs/blob/dev/.changes/readme.md).

File: src/hashes/ternary/kerl/bigint/i384/mod.rs [hunk 2]
```diff
@@ -168,6 +168,12 @@ impl PartialEq for I384<BigEndian, U8Repr> {
 
 impl PartialOrd for I384<BigEndian, U8Repr> {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
```
Comment (semenov-vladyslav) [inline src/hashes/ternary/kerl/bigint/i384/mod.rs:171]: Maybe such changes belong to a different PR?

File: src/signatures/ed25519.rs [hunk 3]
```diff
@@ -136,6 +136,91 @@ impl Hash for PublicKey {
...
+
+impl PublicKeyBytes {
+    pub const LENGTH: usize = PublicKey::LENGTH;
+
+    pub fn verify(&self, sig: &Signature, msg: &[u8]) -> crate::Result<bool> {
+        Ok(self.into_public_key()?.verify(sig, msg))
+    }
+
+    pub fn into_public_key(self) -> crate::Result<PublicKey> {
+        Ok(PublicKey(self.0.try_into().map_err(|_| {
```
Comment (semenov-vladyslav) [inline src/signatures/ed25519.rs]: `self.0.try_into().map_err(...).map(PublicKey)`?

File: tests/ed25519.rs [hunk 4]
```diff
@@ -24,13 +24,13 @@ fn test_zip215() -> crypto::Result<()> {
     for tv in tvs.iter() {
         let mut pkb = [0; PUBLIC_KEY_LENGTH];
         hex::decode_to_slice(tv.public_key, &mut pkb as &mut [u8]).unwrap();
-        let pk = PublicKey::try_from_bytes(pkb)?;
+        let pk = PublicKeyBytes::from_bytes(pkb);
```
Comment (semenov-vladyslav) [inline tests/ed25519.rs]: These changes remove tests for `PublicKey`. Maybe, duplicate it like this:

```rust
let pk = PublicKey::try_from_bytes(pkb)?;
let pkb = PublicKeyBytes::from_bytes(pkb);
...
assert!(pk.verify(&sig, ms));
assert!(pkb.verify(&sig, ms));
```

**Review by semenov-vladyslav — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (DaughterOfMars) [reply Cargo.toml]: right, thanks

File: src/hashes/ternary/kerl/bigint/i384/mod.rs (hunk 2 above)
Comment (DaughterOfMars) [reply src/hashes/ternary/kerl/bigint/i384/mod.rs:171]: just figured I'd fix the clippy issues while I was there

File: tests/ed25519.rs (hunk 4 above)
Comment (DaughterOfMars) [reply tests/ed25519.rs]: hm, ok

**Review by semenov-vladyslav — APPROVED** [review]
Looks good.

**Review by thibault-martinez — APPROVED** [review]
